presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -163,6 +163,15 @@ public void testParsingError()
         assertQueryFails("SELECT foo FROM", "line 1:16: no viable alternative at input.*");
     }
 
+    @Test
+    public void testShowPartitions()
+    {
+        assertQueryFails("SHOW PARTITIONS FROM orders", "line \\S*: Table does not have partition columns: \\S+\\.orders");
+        assertQueryFails("SHOW PARTITIONS FROM orders WHERE orderkey < 10", "line \\S*: Table does not have partition columns: \\S+\\.orders");
+        assertQueryFails("SHOW PARTITIONS FROM orders WHERE invalid_column < 10", "line 1:35: Column 'invalid_column' cannot be resolved");
+        assertQueryFails("SHOW PARTITIONS FROM orders LIMIT 2", "line \\S*: Table does not have partition columns: \\S+\\.orders");
```

Comment:
can you add some tests where table is actually partitioned? You could then check invalid column  (not parition column) usage etc.

Reply:
Such tests are elsewhere. This is only to ensure `SHOW PARTITIONS` responses reasonably for a table, no matter which connector backs the table.